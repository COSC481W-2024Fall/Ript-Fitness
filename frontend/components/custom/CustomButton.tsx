import { TouchableOpacity, Text, StyleSheet, ViewStyle, TextStyle } from 'react-native'

type CustomButtonProps = {
  title: string;
  backgroundColor?: string;
  borderColor?: string;
  borderWidth?: number;
  width?: number,
  textColor?: string;
  fontSize?: number;
  onPress?: () => void;
};

const CustomButton: React.FC<CustomButtonProps> = ({ title, backgroundColor, borderColor, borderWidth, width, textColor, fontSize, onPress }) => {
  const buttonStyle: ViewStyle = {
    backgroundColor: backgroundColor || '#03A696',
    borderColor: borderColor || 'transparent',
    borderWidth: borderWidth !== undefined ? borderWidth : borderColor ? 1 : 0,
    width: width || '70%'
  };

  const textStyle: TextStyle = {
    color: textColor || '#fff',
    fontSize: fontSize || 15,
  };

  return (
    <TouchableOpacity 
      style={[styles.button, buttonStyle]} 
      onPress={onPress}
      activeOpacity={0.7}
    >
      <Text style={[textStyle]}>{title}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
    minHeight: 62,
    margin: 5
  }
});

export default CustomButton