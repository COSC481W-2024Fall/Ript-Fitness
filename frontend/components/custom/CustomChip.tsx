import { useState } from "react";
import { TouchableOpacity, View, StyleSheet, Text } from "react-native";

interface ChipProps {
  label: string;
  value: string;
  selected: boolean;
  onPress: (value: string) => void;
}

const Chip: React.FC<ChipProps> = ({ label, value, selected, onPress }) => {
  return (
    <TouchableOpacity style={[styles.chip, selected && styles.selectedChip]} onPress={() => onPress(value)}>
      <Text style={[styles.chipText, selected && styles.selectedChipText]}>{label}</Text>
    </TouchableOpacity>
  );
};

const CustomChip: React.FC = () => {
  const [selectedChip, setSelectedChip] = useState<string | null>(null);
  
  const handleChipPress = (value: string) => {
    setSelectedChip(value);
  };

  return (
    <View style={styles.container}>
      <Chip label="Upper" value="upper" selected={selectedChip === 'upper'} onPress={handleChipPress} />
      <Chip label="Lower" value="lower" selected={selectedChip === 'lower'} onPress={handleChipPress} />
      <Chip label="Rec" value="rec" selected={selectedChip === 'rec'} onPress={handleChipPress} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  chip: {
    backgroundColor: '#D9D9D9',
    borderRadius: 20,
    padding: 10,
    margin: 5,
    width: '25%',
    // height: 65,
    alignItems: 'center',
  },
  chipText: {
    fontSize: 15,
    // fontWeight: 'bold',
    color: '#1D2526',
  },
  selectedChip: {
    backgroundColor: '#2493BF',
  },
  selectedChipText: {
    color: '#fff',
  }
});

export default CustomChip;