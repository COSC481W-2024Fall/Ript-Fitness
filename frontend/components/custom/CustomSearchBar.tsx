import React from 'react';
import { View, TextInput, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

const SearchBar = () => {
  return (
    <View style={styles.searchContainer}>
     <Ionicons name="search-outline" size={20} color="grey" style={styles.iconStyle} />
      <TextInput
        placeholder="Search"
        style={styles.inputStyle}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  searchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 10,
    padding: 10,
    borderColor: 'grey',
    width: '90%',
    alignSelf: 'center',
    margin: 10, 
  },
  iconStyle: {
    marginRight: 10,
    marginLeft: 10,
  },
  inputStyle: {
    flex: 1, // Ensures the input takes up the remaining space
  },
});

export default SearchBar;