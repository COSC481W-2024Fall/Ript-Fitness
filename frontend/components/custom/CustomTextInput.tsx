import React from 'react';
import { TextInput, TextInputProps, StyleSheet, ViewStyle, View, TouchableWithoutFeedback, Keyboard  } from 'react-native';

type CustomTextInputProps = {
  placeholder: string;
  placeholderTextColor?: string;
  width?: number;
  style?: ViewStyle;
  color?: string;
} & TextInputProps; 


const CustomTextInput: React.FC<CustomTextInputProps> = ({ placeholder, placeholderTextColor, width, style, color, ...props }) => {
  const containerStyle: ViewStyle = {
    width:  width || '100%',
  };

  return (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
      <TextInput
        style={[styles.input, containerStyle, style, { color }]} 
        placeholder={placeholder}
        placeholderTextColor={placeholderTextColor}
        {...props} 
      />
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  input: {
    height: 50,
    backgroundColor: '#EDEDED',
    borderRadius: 25,
    paddingHorizontal: 10,
    paddingLeft: 20,
    marginVertical: 10,
  },
});

export default CustomTextInput;
