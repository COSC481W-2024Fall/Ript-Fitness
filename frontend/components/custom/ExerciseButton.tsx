import { Dimensions, TouchableOpacity, View, type ViewProps } from 'react-native';
import { StyleSheet } from 'react-native';
import { useThemeColor } from '@/hooks/useThemeColor';
import { Ionicons } from '@expo/vector-icons';
import React, { useState } from 'react';
import { ThemedView } from '@/components/ThemedView'
import { ThemedText } from '../ThemedText';

export type ExerciseButtonProps = ViewProps & {
    leftColor?: string;
    title?: string;
    desc?: string;
    children?: React.ReactNode;
    darkColor?: string;
    sideButtons?: any;
    id?: string;
    onLongPress?: any;
    onPressOut?: any;
    isActive?: any
    style?: any
};

export function ExerciseButton({ leftColor, title, desc, children, sideButtons, id, onLongPress, onPressOut, isActive, style, ...otherProps }: ExerciseButtonProps) {
  const toggleExpand = () => {
    if (isExpanded) {
      setExpanded(false)
      setIconStr("chevron-down")
    } else {
      setExpanded(true)
      setIconStr("chevron-up")
    }

    }
  const [iconStr, setIconStr] = useState("chevron-down")

  const [isExpanded, setExpanded] = useState(false);

  return <TouchableOpacity style={[styles.view, style] } onLongPress={onLongPress}  // Long press triggers drag
  disabled={isActive} activeOpacity={1} onPressOut={onPressOut}>

<ThemedView style={styles.fullWidth}>
<ThemedText style={styles.title}>{title}</ThemedText>
{desc && <ThemedView style={styles.innerBar}><ThemedText style={styles.description}>{desc}</ThemedText></ThemedView>}
</ThemedView>

       {isExpanded && (
  <ThemedView style={styles.dropDownDiv}>
    {children}
  </ThemedView>
)}

   <View style={[{backgroundColor:leftColor}, styles.leftColor]}></View>
   

  <TouchableOpacity key='arrow' style={styles.innerButton} onPress={toggleExpand} >
    
  <Ionicons name={iconStr as any} size={30} color="#000000" />

  </TouchableOpacity>
  

  {isExpanded && (<View style={styles.sideButtonsView}>
  {sideButtons?.map((button: {id: string, icon: string, func: any, style:any}) => (
      <TouchableOpacity style={styles.sideButton} key={button.id} onPress={() => button.func(id)}>
    
      <Ionicons name={button.icon as any} style={button.style} size={20} color="#000000" />
    
      </TouchableOpacity>
        ))}
{/*
<TouchableOpacity
      onLongPress={onLongPress}  // Long press triggers drag
      disabled={isActive}        // Disable interaction while dragging
      style={[
        styles.sideButton,
        { backgroundColor: isActive ? '#E8E8E8' : '#E2E2E2', opacity: isActive ? 0.8 : 1}  // Change style while dragging
      ]}
    ><Ionicons name="menu" size={30} color="#000000" /></TouchableOpacity>
*/}
</View>)}

  </TouchableOpacity>;
}

const { width, height } = Dimensions.get('window');
const titleSize = width * 0.06;
const descSize = width * 0.04;

const styles = StyleSheet.create({ 
  fullWidth: {
    width:'100%',
    backgroundColor:'inherit',
    height:'auto',
    alignItems:'center',
    paddingTop:15,
    paddingBottom:15,
  },
  sideButtonsView: {
    position:'absolute',
    bottom:10,
    right:20,
  },
  sideButton: {
    marginTop:8
  },
  title: {
    fontSize:titleSize,
    marginBottom:5,
  },

  description: {
    fontSize:descSize,
    color:'white'
  },
  innerBar: {
    backgroundColor:'black',
    width:'auto',
    paddingTop:'2%',
    paddingBottom:'2%',
    paddingLeft:'5%',
    paddingRight:'5%',
    borderRadius:15,
    marginTop:5,
  },
    view: {
        backgroundColor:'#E2E2E2',
        borderRadius:15,
        marginTop:15,
      },
      innerButton: {
        position: 'absolute',
        right:15,
        top:11,
        height:30,
        width:30,
        //backgroundColor:'blue',
        //color:'white',
        borderTopLeftRadius:15,
        borderBottomLeftRadius:15,
      },
      leftColor: {
        position: 'absolute',
        left:0,
        top:0,
        height:'100%',
        width:18,
        borderTopLeftRadius:15,
        borderBottomLeftRadius:15,
      },
      dropDownDiv: {
        position: 'relative',
        left:30,
        width:'75%',
        paddingTop:'0%',
        height:'auto',
        backgroundColor:'inherit',
        //backgroundColor:'blue',
        paddingBottom:10,
      },



});