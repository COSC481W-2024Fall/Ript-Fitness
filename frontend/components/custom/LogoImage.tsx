import { Image, StyleProp, StyleSheet, ImageStyle, View } from 'react-native'

interface LogoImageProps {
  style?: StyleProp<ImageStyle>;
}


const LogoImage: React.FC<LogoImageProps> = ({ style }) => {
  return (
    // <View style={[styles.logoContainer]}>
    //   <Image 
    //     style={[styles.logo, style]}
    //     source={require('@/assets/images/Ript-Main-Logo1.png')}
    //   />
    // </View>
    <Image 
        style={[styles.logo, style]}
        source={require('@/assets/images/Ript-Main-Logo1.png')}
    />
  );
};

const styles = StyleSheet.create({
  // logoContainer: {
  //   justifyContent: 'center',
  //   alignItems: 'center',
  // },
  logo: {
    width: 300,
    height: 110,
  }
});

export default LogoImage;