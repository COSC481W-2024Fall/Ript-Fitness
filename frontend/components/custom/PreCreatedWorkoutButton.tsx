import { TouchableOpacity, View, Text, StyleSheet } from 'react-native';
import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import { SocialScreenNavigationProp } from '@/app/(tabs)/SocialStack';


const PreCreatedWorkoutButton: React.FC = () => {
    const navigation = useNavigation<SocialScreenNavigationProp >();

    const buttonTitles = ['Push Day 1', 'Pull Day 1', 'Back/bis', 'Legs: Glutes/Ham', 'Push (Summer)', 'Pull (Summer)', 'Legs (Summer)', 'Upper Body', 'Push Day 2', 'Pull Day 2'];
    return (
        <View style={styles.container}>
            {buttonTitles.map((title, index) => (
                <TouchableOpacity 
                    key={index}
                    //title={title}
                    //lastCompleted={"10/07/2024"}
                    onPress={() => navigation.navigate('SocialFeed')}
                    style={styles.button}>
                    <View style={styles.buttonContent}>
                        <Text
                            style={styles.text}
                            numberOfLines={2}
                            ellipsizeMode='tail'>
                        {title}
                        </Text>

                        <View style={styles.textWrapper}>
                            <Text style={styles.dateText}>last completed: 10/07/2024</Text>
                        </View>
                    </View>
                </TouchableOpacity>
            ))}

        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row', 
        flexWrap: 'wrap',
        justifyContent: 'center', 
        gap: 30,
        width: '100%',
    },
    button: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        justifyContent: 'center',
        backgroundColor: 'grey', 
        padding: 5, 
        borderRadius: 10,
        width: 150,
        height: 120,
        position: 'relative',
    }, 
    buttonContent: {
        flexDirection: 'column', 
        position: 'relative',
        height: '100%',
        width: '100%',
    },
    text: {
        textAlign: 'left',
        fontFamily: 'Courier',
        color: 'black',
        fontSize: 15,
        padding: 20,
        position: 'absolute',
        top: -5, 
    },
    textWrapper: {
        position: 'absolute',
        bottom: 6, 
        right: 0, 
        left: 0,
        height: '25%', 
        alignItems: 'center',
        backgroundColor: 'black', 
        padding: 10, 
        borderRadius: 10, 
    }, 
    dateText: {
        position: 'relative',
        color: 'white', 
        fontSize: 7, 
        fontFamily: 'Courier',
        textAlign: 'left',
    },
})

export default PreCreatedWorkoutButton;