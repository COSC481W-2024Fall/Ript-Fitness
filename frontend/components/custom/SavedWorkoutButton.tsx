import React, { useState } from 'react';
import { Dimensions, TouchableOpacity, View, type ViewProps, StyleSheet } from 'react-native';
import { ThemedText } from '../ThemedText';

export type SavedWorkoutButtonProps = ViewProps & {
    key: any;
    title: string;
    level: string; 
    time: number; 
    // exercises: [{
    //     exercise: string;
    //     sets: number;
    //     reps: number;
    // }];
    children?: React.ReactNode;
    darkColor?: string;
    id?: string;
    onLongPress?: any;
    onPress?: any;
    isActive?: any
    style?: any
};


export function SavedWorkoutButton({title, level, time, children, id, onLongPress, onPress, isActive, style, ...otherProps }: SavedWorkoutButtonProps) {
  
    return (
        <View style={styles.container}>
            <TouchableOpacity
                onPress={onPress}
                style={styles.button}
            >  
                <ThemedText style={styles.text}>{title}</ThemedText>

                <View style={styles.textWrapper}>
                    <ThemedText style={styles.dateText}>{level}: {time} min</ThemedText>
                </View>
            </TouchableOpacity>
        </View>
    )
}
  const { width, height } = Dimensions.get('window');
  const titleSize = width * 0.06;
  const descSize = width * 0.04;

  const styles = StyleSheet.create({ 
    container: {
        flexWrap: 'wrap',
        flexDirection: 'row', 
        justifyContent: 'space-evenly', 
        gap: 15,
        width: width,
    },
    button: {
        flexWrap: 'wrap',
        justifyContent: 'flex-end',
        backgroundColor: 'white', 
        padding: 5, 
        margin: 5,
        borderRadius: 20,
        borderColor: 'grey',
        borderWidth: 1, 
        width: width/2-20,
        height: 120,
        alignContent: 'center',
    }, 
    text: {
        color: 'black',
        fontWeight: 'bold',
        fontSize: 15,
        padding: 20,
        position: 'absolute',
        top: -5, 
    },
    textWrapper: {
        alignItems: 'center',
        backgroundColor: '#2493BF',
        padding: 10, 
        borderRadius: 40, 
        width: '80%',
    }, 
    dateText: {
        color: 'black', 
        fontSize: 10, 
    },
  });
  

