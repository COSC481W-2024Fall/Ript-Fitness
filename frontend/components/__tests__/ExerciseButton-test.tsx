import * as React from 'react';
import { render } from '@testing-library/react-native';
import { StyleSheet, View } from 'react-native';
import { ThemedText } from '../ThemedText';
import { ExerciseButton } from '../custom/ExerciseButton';

it(`ExerciseButton renders correctly`, () => {
  const workouts = [
    { id: '1', name: 'Push-Ups', description: "5 Sets", values: ["10", "9", "8", "8", "9"], valueTypes: "Reps", color: "#F2846C" }
  ];

  const sideButtons = [
    { id: '1', icon: 'pencil', func: undefined },
    { id: '2', icon: 'arrow-up', func: undefined, style: styles.viewButton },
  ];

  const item = workouts[0]; // Access the first workout
  const { tree } = render(
    <ExerciseButton
      id={item.id}
      leftColor={item.color}
      title={item.name}
      desc={item.description}
      isActive={false}
      style={styles.myWidth}
      sideButtons={sideButtons}
    >
      {item.values.map((value, index) => (
        <View key={index} style={styles.rowItem}>
          <ThemedText style={styles.floatLeft}>{value}</ThemedText>
          <ThemedText style={styles.floatRight}>{item.valueTypes}</ThemedText>
        </View>
      ))}
    </ExerciseButton>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

const styles = StyleSheet.create({
  viewButton: {
    transform: [{ rotate: '90deg' }],
  },
  rowItem: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  floatLeft: {
    textAlign: 'left',
  },
  floatRight: {
    textAlign: 'right',
  },
  myWidth: {
    width: '100%',
  }
});
