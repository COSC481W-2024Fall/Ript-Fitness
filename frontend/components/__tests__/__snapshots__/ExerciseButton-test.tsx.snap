// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#E2E2E2",
      "borderRadius": 15,
      "marginTop": 15,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "backgroundColor": "inherit",
          "height": "auto",
          "paddingBottom": 15,
          "paddingTop": 15,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 45,
            "marginBottom": 5,
          },
        ]
      }
    >
      Push-Ups
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "backgroundColor": "black",
            "borderRadius": 15,
            "marginTop": 5,
            "paddingBottom": "2%",
            "paddingLeft": "5%",
            "paddingRight": "5%",
            "paddingTop": "2%",
            "width": "auto",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
      >
        5 Sets
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#F2846C",
        },
        {
          "borderBottomLeftRadius": 15,
          "borderTopLeftRadius": 15,
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 18,
        },
      ]
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderBottomLeftRadius": 15,
        "borderTopLeftRadius": 15,
        "height": 30,
        "opacity": 1,
        "position": "absolute",
        "right": 15,
        "top": 11,
        "width": 30,
      }
    }
  >
    <Ionicons
      color="#000000"
      name="chevron-down"
      size={30}
    />
  </View>
</View>
`;
